### PR TITLE
[CHI-79] Add steps component

### PR DESCRIPTION
**Pull request by davidojedacl**
@mattnickles @jllr

### CODE REVIEW

File: src/chi/components/steps/steps.scss [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        & a {
+          color: $brand-color;
+        }
+
+        &:before {
+          box-shadow: 0 0 0 0.125rem $brand-color;
+        }
+
+        &:after {
+          background-color: $muted-color;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: all references to muted-color in the circle shapes and bar should be replaced with the color value grey-40 (#b0b0b0)

File: src/chi/components/steps/steps.scss [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+          &:after {
+            background-color: $muted-color;
+          }
+        }
+      }
+    }
+
+    & a {
+      color: $muted-color;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Active step link is $brand-color, completed step link is $text-color, and the not yet completed step links are $muted-color

image

File: src/chi/components/steps/steps.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        z-index: -1;
+      }
+
+      &:last-child {
+        &:after {
+          display: none;
+        }
+      }
+
+      &.-active {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: The white in the active circle is rendering slightly larger than the mock

Current:
image
Expected:
image

**Comment by mattnickles** [comment]
Can we also address responsive behavior? @jllr 

image

File: src/chi/components/steps/steps.scss [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        &:before {
+          box-shadow: 0 0 0 0.125rem $brand-color;
+        }
+
+        &:after {
+          background-color: set-color(grey, 40);
+        }
+
+        & ~ li {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: 👍  Nice!

File: test/chi/components/steps.pug [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+title: Steps
+---
+
+- var steps = ['Create account', 'Choose number', 'Add seats', 'Choose services', 'Checkout'];
```
Comment (mattnickles) [inline test/chi/components/steps.pug]: For the test, can we make this more generic and just use "Step 1" "Step 2", etc. for labels

File: src/chi/components/steps/steps.scss [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+.chi {
+
+  .a-steps {
+    display: table;
+    padding: 0;
+    table-layout: fixed;
+    width: 100%;
+
+    & > li {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: We need to set line-height to 1rem for when the labels wrap

File: src/chi/components/steps/steps.scss [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+      &:before {
+        background-color: $text-white;
+        border-radius: 50%;
+        content: '\25CF';
+        display: block;
+        font-size: 1.85rem;
+        height: 1.125rem;
+        line-height: 1rem;
+        margin: 0 auto 0.25rem;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: After we alter the line-height on the labels, it's going to push the labels up. We can add that space back by increasing the bottom margin here to `0.50rem`

File: src/chi/components/steps/steps.scss [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+          &:after {
+            background-color: set-color(grey, 40);
+          }
+        }
+      }
+    }
+
+    & a {
+      color: $text-color;
```
Comment (jllr) [inline src/chi/components/steps/steps.scss]: @davidojedacl remove text-decoration when hover or change the html tag to span or p

Comment (mattnickles) [reply src/chi/components/steps/steps.scss]: anchor tag is fine, let's just set `text-decoration: none;`

Comment (mattnickles) [reply src/chi/components/steps/steps.scss]: @davidojedacl actually, can we set all links in the steps component to `text-decoration: none;` ?

Comment (davidojedacl) [reply src/chi/components/steps/steps.scss]: ok

File: src/chi/components/steps/steps.scss [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    & > li {
+      color: $brand-color;
+      display: table-cell;
+      font-size: $text;
+      line-height: 1rem;
+      position: relative;
+      text-align: center;
+      z-index: 0;
+
+      &::before {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Change to:

```
&::before {
  background-color: $brand-color;
  border: 0.125rem solid $text-white;
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  line-height: 1rem;
  margin: 0 auto 0.5rem;
  width: 1rem;
}
```

File: src/chi/components/steps/steps.scss [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        & ~ li {
+          color: set-color(grey, 40);
+
+          & a {
+            color: $muted-color;
+          }
+
+          &::before {
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Change to:
```
&::before {
  background-color: set-color(grey, 40);
  border-color: $text-white;
}
```

File: src/chi/components/steps/steps.scss [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+          &::after {
+            background-color: set-color(grey, 40);
+          }
+        }
+      }
+    }
+
+    & a {
+      color: $text-color;
+      text-decoration: none;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: Add:
```
&:hover {
  color: $brand-color;
}
```

File: src/chi/components/steps/steps.scss (hunk 9 above)
Comment (davidojedacl) [reply src/chi/components/steps/steps.scss]: I changed with this code and now not is horizontal centered i think. what do you think? It 
screen shot 2018-11-21 at 9 52 24 am
is ok?

Comment (mattnickles) [reply src/chi/components/steps/steps.scss]: @davidojedacl Good catch. Can we fix this while ensuring the circle is rendered as a shape versus a web font?

File: src/chi/components/steps/steps.scss [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+        width: 1rem;
+      }
+
+      &::after {
+        background-color: $brand-color;
+        content: '';
+        height: 0.125rem;
+        left: 50%;
+        position: absolute;
+        top: 0.5rem;
```
Comment (mattnickles) [inline src/chi/components/steps/steps.scss]: @davidojedacl @jllr to fix the vertical alignment of the circle, we need to adjust the `::after`, not the `::before`. the top value here should be `top: .4375rem;` and that should do it.

**Pull request merged**